static/src/components/setting/index.vue

```diff
@@ -0,0 +1,326 @@
+                <th style="border-right: 1px ridge silver; text-align: center">
+                  All Users
+                </th>
+                <th style="text-align: center">
+                  Assigned Users
+                </th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr>
+                <td style="border-right: 1px ridge silver; width:50%">
+                  <input
+                    v-model="search"
+                    class="form-group input-sm"
+                    type="text"
+                    placeholder="Search for name.."
+                    style="min-width: 250px"
+                    @keyup.enter="filter"
+                  >
+                  <div class="col-sm-12 scroll">
+                    <div
+                      v-for="u in searchResult"
+                      id="users"
+                      :key="u.id"
+                      class="row"
+                      :value="u"
+                      @click="add($event, u)"
+                    >
+                      <p
+                        v-if="Object.keys(assignee).includes(u.id)"
+                        style="color:teal"
+                      >
+                        {{ u.fullname }}
+                      </p>
+                      <p v-else>
+                        {{ u.fullname }}
+                      </p>
+                    </div>
+                  </div>
+                </td>
+                <td>
+                  <div class="col-sm-12 scroll">
+                    <div
+                      v-for="u in assignee"
+                      id="users"
+                      :key="u.name"
+                      class="row"
+                      :value="u.field"
+                      @click="remove($event, u.id)"
+                    >
+                      <p> {{ u.fullname }}</p>
+                    </div>
+                  </div>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </div>
+      <div>
+        <button
+          class="btn btn-sm btn-primary"
+          @click="save"
+        >
+          Save
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Vue from 'vue';
+export default {
+
+  props: {
+    csrfToken: {
+      type: String,
+      default: null
+    },
+    dataAccess: {
+      type: Array,
+      default: function () { return []; }
+    },
+    assignUsers: {
+      type: Array,
+      default: function () { return []; }
+    },
+    allUsers: {
+      type: Array,
+      default: function () { return []; }
+    },
+    isPrivate: {
+      type: Boolean,
+      default: true
+    },
+    externalConfig: {
+      type: Object,
+      default: () => ({ gigwork_poller: { target_bucket: null } })
+    }
+  },
+
+  data () {
+    return {
+      bcos: this.externalConfig.gigwork_poller.target_bucket,
+      assignee: {},
+      users: {},
+      L1: false,
```

Comment:
currently levels are defined in the server's configuration file, not hard-coded

Reply:
do you mean we should read access levels from configuration file in case there're changes on  levels?

Reply:
The code reads the levels from the configuration, so one can spin up a server with any access level they want. This page would only make sense for a particular choice of the levels (L1 L2 L3 L4)

Reply:
agree, will fix in next commit